src/data/lovelace.ts

```diff
@@ -60,95 +62,47 @@ export const fetchConfig = (
     force,
   });
 
-export const migrateConfig = (hass: HomeAssistant): Promise<void> =>
-  hass.callWS({
-    type: "lovelace/config/migrate",
-  });
-
 export const saveConfig = (
   hass: HomeAssistant,
-  config: LovelaceConfig | string,
-  format: "json" | "yaml"
+  config: LovelaceConfig
 ): Promise<void> =>
   hass.callWS({
     type: "lovelace/config/save",
     config,
-    format,
   });
 
-export const getCardConfig = (
-  hass: HomeAssistant,
-  cardId: string
-): Promise<string> =>
-  hass.callWS({
-    type: "lovelace/config/card/get",
-    card_id: cardId,
-  });
+async function notImplemented() {
+  alert("Not implemented yet...");
+}
 
 export const updateCardConfig = (
   hass: HomeAssistant,
-  cardId: string,
-  config: LovelaceCardConfig | string,
-  format: "json" | "yaml"
-): Promise<void> =>
-  hass.callWS({
-    type: "lovelace/config/card/update",
-    card_id: cardId,
-    card_config: config,
-    format,
-  });
+  config: LovelaceCardConfig
+): Promise<void> => notImplemented();
 
 export const deleteCard = (
```

Comment:
We should mutate these functions to be like this:

```
export const deleteCard = (config: LovelaceCardConfig, viewIndex, cardIndex): LovelaceCardConfig;
```

Reply:
And then the caller can route the result to `updateCardConfig`. Will make it easy to test the functions and reason about it too.